rmf_traffic/include/rmf_traffic/Trajectory.hpp

```diff
@@ -43,270 +42,115 @@ class Trajectory
-
-    /// Change the profile of this Trajectory Segment
-    ///
-    /// \param[in] new_profile
-    ///   The new profile for this Trajectory Segment.
-    Segment& set_profile(ConstProfilePtr new_profile);
-
     /// Get the intended physical location of the robot at the end of this
-    /// Trajectory Segment.
+    /// Trajectory Waypoint.
     ///
     /// This is a 2D homogeneous position. The first two values in the vector
     /// are x and y coordinates, while the third is rotation about the z-axis.
-    Eigen::Vector3d get_finish_position() const;
+    Eigen::Vector3d position() const;
 
     /// Set the intended physical location of the robot at the end of this
-    /// Trajectory Segment.
+    /// Trajectory Waypoint.
     ///
     /// This is a 2D homogeneous position. The first two values in the vector
     /// are x and y coordinates, while the third is rotation about the z-axis.
     ///
     /// \param[in] new_position
-    ///   The new finishing position for this Trajectory Segment.
-    Segment& set_finish_position(Eigen::Vector3d new_position);
+    ///   The new finishing position for this Trajectory Waypoint.
+    Waypoint& position(Eigen::Vector3d new_position);
 
     /// Get the intended velocity of the robot at the end of this Trajectory
-    /// Segment.
+    /// Waypoint.
     ///
     /// This is a 2D homogeneous position. The first two values in the vector
     /// are x and y velocities, while the third is rotational velocity about the
     /// z-axis.
-    Eigen::Vector3d get_finish_velocity() const;
+    Eigen::Vector3d velocity() const;
 
     /// Set the intended velocity of the robot at the end of this Trajectory
-    /// Segment.
+    /// Waypoint.
     ///
     /// This is a 2D homogeneous position. The first two values in the vector
     /// are x and y coordinates, while the third is rotation about the z-axis.
     ///
     /// \param[in] new_velocity
-    ///   The new finishing velocity for this Trajectory Segment.
-    Segment& set_finish_velocity(Eigen::Vector3d new_velocity);
+    ///   The new finishing velocity for this Trajectory Waypoint.
+    Waypoint& velocity(Eigen::Vector3d new_velocity);
 
-    /// Get the time that this Trajectory Segment is meant to finish.
-    Time get_finish_time() const;
+    /// Get the time that this Trajectory Waypoint is meant to finish.
+    Time time() const;
 
-    /// Change the finish time of this Trajectory Segment. Note that this
-    /// function will only affect this segment, and may cause this Segment to be
+    /// Change the finish time of this Trajectory Waypoint. Note that this
+    /// function will only affect this waypoint, and may cause this waypoint to be
     /// reordered within the Trajectory.
     ///
-    /// To change the finish time for this segment while preserving the relative
+    /// To change the finish time for this waypoint while preserving the relative
     /// times of all subsequent Trajectory segments, use adjust_finish_times()
```

Comment:
Should this also be waypoint? 
```suggestion
    /// times of all subsequent Trajectory **waypoints**, use adjust_finish_times()
```